openeo/rest/connection.py

```diff
@@ -211,6 +212,37 @@ def _get_refresh_token_store(self) -> RefreshTokenStore:
             self._refresh_token_store = RefreshTokenStore()
         return self._refresh_token_store
 
+    def list_auth_providers(self) -> list[dict]:
+        providers = []
+        cap = self.capabilities()
+
+        # Add OIDC providers
+        oidc_path = "/credentials/oidc"
+        if cap.supports_endpoint(oidc_path, method="GET"):
+            try:
+                data = self.get(oidc_path, expected_status=200).json()
+                if isinstance(data, dict):
+                    for provider in data.get("providers", []):
+                        provider["type"] = "oidc"
+                        providers.append(provider)
+            except OpenEoApiError:
+                pass
+
+        # Add Basic provider
+        basic_path = "/credentials/basic"
+        if cap.supports_endpoint(basic_path, method="GET"):
+            providers.append(
+                {
+                    "id": uuid.uuid4().hex,
+                    "issuer": self.build_url(basic_path),
+                    "type": "basic",
+                    "title": "Internal",
```

Comment:
I'd rather call this "Basic Auth".
"Internal" could also be useful for one of the OIDC provider options (e.g. for special use cases where you have a special, VPN-guarded OIDC provider for developers/testers/machine-to-machine-auth/...)

Reply:
@niebl Depending on the outcome, you may need to reflect this in the QGIS plugin code.